src/Adapter/Shop/MaintenanceConfiguration.php

```diff
@@ -61,24 +51,40 @@ public function getConfiguration()
      */
     public function updateConfiguration(array $configuration)
     {
-        if ($this->validateConfiguration($configuration)) {
-            $this->configuration->set('PS_SHOP_ENABLE', $configuration['enable_shop']);
-            $this->configuration->set('PS_MAINTENANCE_IP', $configuration['maintenance_ip']);
-            $this->configuration->set('PS_MAINTENANCE_TEXT', $configuration['maintenance_text'], null, ['html' => true]);
+        $shopConstraint = null;
+        if (!$this->shopContext->isAllShopContext()) {
+            $configuration = $this->removeDisabledFields($configuration);
+            $contextShopGroup = $this->shopContext->getContextShopGroup();
+            $contextShopId = $this->shopContext->getContextShopID();
+            $contextShopId = (int) $contextShopId > 0 ? $contextShopId : null;
+
+            $shopConstraint = new ShopConstraint(
+                $contextShopId,
+                $contextShopGroup->id
+            );
+        }
+
+        if (isset($configuration['enable_shop'])) {
+            $this->configuration->set('PS_SHOP_ENABLE', $configuration['enable_shop'], $shopConstraint);
+        }
+        if (isset($configuration['maintenance_ip'])) {
+            $this->configuration->set('PS_MAINTENANCE_IP', $configuration['maintenance_ip'], $shopConstraint);
+        }
+        if (isset($configuration['maintenance_text'])) {
+            $this->configuration->set('PS_MAINTENANCE_TEXT', $configuration['maintenance_text'], $shopConstraint, ['html' => true]);
         }
 
         return [];
     }
 
     /**
-     * {@inheritdoc}
+     * @param array $configuration
+     *
+     * @return bool
      */
-    public function validateConfiguration(array $configuration)
+    public function validateConfiguration(array $configuration): bool
     {
-        return isset(
-            $configuration['enable_shop'],
-            $configuration['maintenance_ip'],
-            $configuration['maintenance_text']
-        );
+        // TODO: Implement validateConfiguration() method.
```

Comment:
So there is no more validation? Does it mean you plan on re-implementating it in future PR?

Reply:
well the validation consisted of an isset, but with the multistore implementation fields can actually be unset... but still this method is mandatory...

Do you have validation ideas ?

Reply:
maybe I could check that at least one field is set

Reply:
So the previous validation was blocking because in multi shop context not all fields are sent because they might be disabled and absent from the request?

Reply:
yep, I just added a soft validation that only checks that given fields are in the list of expected fields